homeassistant/components/todoist/calendar.py

```diff
@@ -188,12 +189,13 @@ def handle_new_task(call):
             due_date = dt.parse_datetime(call.data[DUE_DATE])
             if due_date is None:
                 due = dt.parse_date(call.data[DUE_DATE])
+                if due is None:
+                    raise ValueError(f"Invalid due_date: {call.data[DUE_DATE]}")
                 due_date = datetime(due.year, due.month, due.day)
             # Format it in the manner Todoist expects
             due_date = dt.as_utc(due_date)
             date_format = "%Y-%m-%dT%H:%M%S"
-            due_date = datetime.strftime(due_date, date_format)
-            _due["date"] = due_date
+            _due["date"] = datetime.strftime(due_date, date_format)
```

Comment:
`mypy` was complaining about storing a string when the original type hint was deduced as `datetime | None`.
I dropped the intermediate variable to store the value directly in the dict.